server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDataStreamsService.java

```diff
@@ -159,4 +242,31 @@ private static IndexAbstraction validateIndex(Metadata metadata, String indexNam
         return index;
     }
 
+    /**
+     * A cluster state update task that consists of the cluster state request and the listeners that need to be notified upon completion.
+     */
+    static class ModifyLifecycleTask extends AckedBatchedClusterStateUpdateTask {
```

Comment:
Again, maybe just a personal preference (feel free to ignore:) ) but maybe this would be better named `UpdateDataLifecycleTask`?

Reply:
No preference, I was following the existing style of the class. `Update` feels more familiar so I will change it.